unit-tests/syncer/test-ts-eof.py

```diff
@@ -0,0 +1,150 @@
+# License: Apache 2.0. See LICENSE file in root directory.
+# Copyright(c) 2021 Intel Corporation. All Rights Reserved.
+
+import pyrealsense2 as rs
+from rspy import log, test
+import sw
+
+
+# The timestamp jumps are closely correlated to the FPS passed to the video streams:
+# syncer expects frames to arrive every 1000/FPS milliseconds!
+sw.fps_c = sw.fps_d = 60
+sw.init()
+
+import tempfile, os
+temp_dir = tempfile.TemporaryDirectory( prefix = 'recordings_' )
+filename = os.path.join( temp_dir.name, 'rec.bag' )
+recorder = rs.recorder( filename, sw.device )
+
+sw.start()
+
+#############################################################################################
+#
+test.start( "Init" )
+
+# It can take a few frames for the syncer to actually produce a matched frameset (it doesn't
+# know what to match to in the beginning)
+
+sw.generate_depth_and_color( frame_number = 0, timestamp = 0 )
+sw.expect( depth_frame = 0 )                          # syncer doesn't know about color yet
+sw.expect( color_frame = 0, nothing_else = True )     # less than next expected of D
```

Comment:
`less than next expected of D` , 
1. D = depth?
2. What do you mean by less than next expected. I didn't understand the comment

Reply:
You need to understand the syncer: every stream has a time at which we expect the next frame to arrive. This is called the "next expected".

Reply:
I know what next expected means :)
I meant that a comment like "Since the next expected time for a depth frame is less than the time the color frame arrive, the syncer will not wait for it and output the color frame"

Will be more verbose without knowing the inside internal parameters of the syncer.